### PR TITLE
Update `regtests-promote` to promote the help texts

### PR DESCRIPTION
**Pull request by antalsz**
This requires matching versions of `cmdliner` with the CI, namely 1.2.0

### CODE REVIEW

**Comment by goldfirere** [comment]
I still don't understand what this is trying to do. Is there a problem for which this is the solution? Or is this just for general hygeine? The comment says something about matching versions of `cmdliner`, but how does one ensure that? And what has to match what? The PR itself doesn't say anything about `cmdliner` or versions.

Sorry to be asking so many questions about such a small patch -- I just don't have the context I need to be able to say whether this is a good idea.

**Comment by antalsz** [comment]
These are all very good questions!  Let me me a bit more loquacious:

* There are a couple of files in the repository that contain the output of the help text for `ocamlformat` (e.g., `ocamlformat --help`).
* The CI checks that these files are consistent with the result of running the help commands.
* The command I've added to `regtests-promote` promotes these help texts; it's running the appropriate `dune` command to regenerate them.
* Currently, it's very easy to end up in a position where you've done all your work and then CI is unhappy because you didn't update the help texts; this change automates that.
* The version of `cmdliner` used locally and remotely has to match because that affects some of the standard verbiage generated in the help text; I solved this by updating locally.

Does that help?

**Comment by goldfirere** [comment]
Yes -- thanks! Sounds very reasonable. But then maybe there should be an addition to HACKING.md that says to install `cmdliner.1.2` or whatever the correct `opam` argument is.

Also, should I say `make regtests-promote` at some point? I've never done this. So if there's any change to the update instructions, please put those in the HACKING.md doc, too.

Thanks!

File: HACKING.jst.md [hunk 1]
```diff
@@ -172,14 +176,10 @@ is:
...
-the generated `tests/passing/dune.inc` file to run your new tests. Run
-`dune promote` to update `dune.inc`. This will *not* accept your new tests -- it
-just allows you to run your new tests.
-
-Then, run `dune test` again to actually run your tests. You will see any changes
-necessary to make the tests pass. You can run `dune promote` to accept those
-changes.
+To test you can run `make test`, which will run the `dune` tests (via `make
+regtest`) and the formatting consistency check (via `make fmt`).  To promote the
+`dune` tests and the `ocamlformat --help` output, run `maketest
```
Comment (goldfirere) [inline HACKING.jst.md:181]: `maketest`?

File: HACKING.jst.md [hunk 2]
```diff
@@ -172,14 +176,10 @@ is:
   when running your test (uncommon) write that output to
   `tests/passing/tests/foo.ml.err`.
 
-Now, run `dune test`. It will discover your new file and suggest edits to
-the generated `tests/passing/dune.inc` file to run your new tests. Run
-`dune promote` to update `dune.inc`. This will *not* accept your new tests -- it
```
Comment (goldfirere) [inline HACKING.jst.md]: Does `make test` really cover all of this? That is, I had to promote *twice* when adding a new test: once to accept the changes to `dune.inc` and the next to actually accept the new tests. I haven't experimented, but it looks like `make test` will not quite do this all in one go. Regardless, I think it's helpful to learn here that `dune.inc` is generated and how it gets updated.

**Comment by goldfirere** [comment]
Closing as abandoned. If anyone thinks this is worthwhile to resurrect, please do so!

**Pull request closed without merging**